go-controller/pkg/metrics/master.go

```diff
@@ -223,20 +225,43 @@ var metricPortBindingUpLatency = prometheus.NewHistogram(prometheus.HistogramOpt
 	Buckets:   prometheus.ExponentialBuckets(.01, 2, 15),
 })
 
-var registerMasterMetricsOnce sync.Once
-var startMasterMetricUpdaterOnce sync.Once
+// The nb_cfg metrics: Currently, nb_cfg is a random serial number that we increment every 30 seconds.
+// northd copies this to SBDB table SB_Global, then ovn-controller reads this and writes it to a chassis-specific
+// table Chassis_Private in SBDB as well
+
+// metricNbcfgNbdb is the value of nb_cfg we write to NBDB
+// There is an equivalent metric for the value in SBDB
+var metricNbcfgNbdb = prometheus.NewGauge(prometheus.GaugeOpts{
+	Namespace: MetricOvnkubeNamespace,
+	Subsystem: MetricOvnkubeSubsystemMaster,
+	Name:      "nb_cfg_nbdb",
+	Help:      "The value of nb_cfg in OVN Northbound database NB_Global table",
+})
+
+var metricNbcfgChassis = prometheus.NewDesc(
+	prometheus.BuildFQName(MetricOvnkubeNamespace, MetricOvnkubeSubsystemMaster, "nb_cfg_chassis"),
+	"The value of nb_cfg in OVN Southbound database Chassis_Private table as reported by each OVN-controller",
+	[]string{"node"}, nil,
+)
+
+var (
+	registerMasterMetricsOnce    sync.Once
+	startMasterMetricUpdaterOnce sync.Once
+	// nbCfgValue is an arbitrary number written to NB_Global tables nb_cfg field
+	nbCfgValue int
```

Comment:
Instead of holding this internally as a var shouldn't we just be incrementing what's already set in nbGlobal?

Reply:
I think it doesn't matter as long as we continuously increment. We don't care about what is written there as long as it isn't static and doesn't repeat frequently.